src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -1440,8 +1440,9 @@ void AbstractWriteLog<I>::alloc_and_dispatch_io_req(C_BlockIORequestT *req)
       std::lock_guard locker(m_lock);
       dispatch_here = m_deferred_ios.empty();
       // Only flush req's total_bytes is the max uint64
-      if ((req->image_extents_summary.total_bytes ==
-          std::numeric_limits<uint64_t>::max())) {
+      if (req->image_extents_summary.total_bytes ==
+          std::numeric_limits<uint64_t>::max() &&
+          dynamic_cast<C_FlushRequestT *>(req)->internal == true) {
```

Comment:
Since only `C_FlushRequest` requests are supposed to have `total_bytes == max()` because of `new C_FlushRequestT(*this, flush_begins, Extents({whole_volume_extent()})`, perhaps this should be a `static_cast` to avoid the performance penalty?